### DISCUSSION TITLE
How to change the default shell to fish?

### DISCUSSION THREAD

**Discussion by moritz1000** [ghostty-org/ghostty#3507] [Q&A] [answered]
Hello,

I am running Arch Linux and using Fish as an **interactive shell** exclusively in my terminal emulators. Fish is **not set as my system shell**.

I recently installed Ghossty and looked at the [Ghossty documentation](https://ghostty.org/docs/config). However, I couldn't find an obvious way to configure Ghossty to use Fish as the shell.

The "Shell Integration" section mentioned in the documentation doesn't seem to address what I'm looking for in this case.
___

On **Alacritty** I can do this via `alacritty.toml`:
```toml
[terminal]
[terminal.shell]
program = "/usr/bin/fish"
```

On **Konsole** via the GUI:
![Image](https://github.com/user-attachments/assets/fc1360e7-08fe-4bc4-9f45-9ca82909249d)

On **kitty** via `kitty.conf`:
```
shell /usr/bin/fish
```

___

How can I configure Ghossty to use Fish as the shell?  
Thank you for your help!

**Comment by zzhaolei** [comment, marked as answer]
`~/.config/ghostty/config`:
`command = /usr/local/bin/fish --login --interactive`

**Reply by maulvi-zm** [reply]
I am doing this on macOS but It's causing all of my shells can't detect binaries in PATH correctly. For example my fish shell can't detect starship and fastfetch. Is there anything I can do to fix this? I installed fish using homebrew and added the following line to my configuration file.

```
command = /opt/homebrew/bin/fish
```

Below is the error. The second image shows the issue after I added the command configuration.

<img width="912" alt="Screenshot 2024-12-28 at 13 56 29" src="https://github.com/user-attachments/assets/b1ad5ab4-df48-4c70-8653-9202efbd282b" />
<img width="912" alt="Screenshot 2024-12-28 at 13 57 42" src="https://github.com/user-attachments/assets/bf4291d9-3e66-4eeb-bd92-85148e07efae" />

**Reply by zzhaolei** [reply]
Apple silicon? You need to add `/opt/homebrew/bin/brew shellenv | source` to the beginning of `config.fish`.

**Reply by jmg-duarte** [reply]
OMG, here's a virtual kiss for you @zzhaolei 💋

I was going INSANE, thank you so much

**Reply by Biangming4572398** [reply]
To whoever is searching this problem in 2026, if `~/.config/ghostty/config` is not found on your device it is because it is now `~/.config/ghostty/config.ghostty` the old one is still recognised by ghostty so you could ideally make a new config file then write in it.

**Comment by moritz1000** [comment]
@zzhaolei Thank you!

I think this feature deserves more prominent documentation and should not be burried in https://ghostty.org/docs/config/reference#command since everyone who is using a different interactive shell requires this information.

At least it would be friendlier for beginners - especially since the documentation has no search function.

**Comment by el-hult** [comment]
As a reminder for people like me: 

to reload the config you must properly quit (e.g. ⌘Q) Ghostty and open it again. Just closing all terminal windows does not stop the program.

**Reply by amanduggal** [reply]
Using reload config `Command + Shift + , `  keyboard shortcut also work. You don't need to quit the terminal. At least it works for me.

**Comment by deadcoder0904** [comment]
Yeah, this should be in documentation. Even AI couldn't help. I'm on Mac M4 & here's what worked:

1. Open Ghostty, press `CMD+,` & enter this at the top of the config:

```bash
shell-integration = fish
command = /opt/homebrew/bin/fish --login --interactive
```

The config is located at `~/.config/ghostty/config` but you can just do the above to find it.

2. And then close the Ghostty terminal and load it again.

Then fish loads in Ghostty.

**Reply by 00-kat** [reply]
> Yeah, this should be in documentation.

https://ghostty.org/docs/config/reference#command.  Pretty easy to find by searching for `shell` in that page.

**Reply by deadcoder0904** [reply]
@00-kat not as a first-time user. the docs should be understood by newbies. its a very high-bar that very docs ever meet like react.dev or vuejs.org even tho ik this is targeted at developers but still.

that's why i left the comment so i can refer again in years if i ever change my mac.

**Reply by 00-kat** [reply]
What would be your suggestion then?  A page detailing many common configuration options…?

**Reply by deadcoder0904** [reply]
also, it needs to take SEO into consideration. i landed on this post while searching ghostty + fish. i think you guys need to copy how warp.dev does it as they have pretty rad SEO. See https://www.warp.dev/terminus/linux-wildcards

Ideally, as a user, I'd expect what I wrote in a nice slug that ranks on Google with a mini-guide like how bun.sh has. See https://bun.sh/guides and check how JSON file guide is written. Some words and a code-block so u can copy-paste with filename at top. It is entering SEO/marketing territory tho rather than plain ol' docs but will increase adoption for sure.

**Reply by deadcoder0904** [reply]
> What would be your suggestion then? A page detailing many common configuration options…?

Oh yes, the best docs imo are:

1. Vercel's many examples (that's how Next.js grew popular + Vercel's AI SDK)
2. Tailwind CSS (see how it shows stuff visually)
3. React.dev (recent site is gem)
4. Vuejs.org (was a gem since v1 since Chinese devs don't have English as a first language so they make it very clear)

Like I wanted to see Theme images but it was so cumbersome to visually see it on the website when I enter `ghostty theme` in Google. Mostly because I like to visually see it. Tailwind CSS does a great job at this. Lots of images will definitely take effort but will 100% take this project to the next level. I love how snappy it is with Fish combo as I'm using it for AI Terminal work but I didn't install Fish until today because I loved how the terminal is so fast.

Also a basic config should be generated by default. Like vesper is a gold theme so users can just start editing (idk if its already commented)

Here's how I search for things:

`ghostty <keyword` so

1. `ghostty theme` (/ghostty-theme should be the slug) where I expect to see lots of themes like Tailwind CSS where I search `tailwind colors` & it shows a nice page (here's some SEO tips bdw u can apply - https://startupspells.com/p/submagic-seo-playbook-free-tools-traffic)
2. `ghostty fish` (/ghostty/shells/fish should be the slug where `/ghostty/shells` contains all slugs) where I want to see how to setup fish in <3 mins with quick copy-paste (would love it if its actually a one-liner that just allows me to paste in the terminal & it just works)

Also, llms.txt plus bun.sh has lots of copy markdown/download markdown buttons for LLMs on its docs (ex: https://bun.sh/docs/runtime/shell)

TL;DR:

Think of a user would search for. Use SEO best-practices (very simple as linked above). Interlink related pages. And lots of images (GIFs are even better or videos but they require more bandwidth). That's it.

GOAT Terminal but poor docs (for now especially I'm a dev but vibe-coders would love this terminal too thanks to its speed). Used to be on WARP but the speed of this blows WARP away. Hope this helps :)

**Reply by rhodes-b** [reply]
> Even AI couldn't help

As expected ghostty came out in december of 2024 as such any AI that exists has absolutely zero to a small amount of knowledge of what ghostty is (even if they are allowed to search the web)

searching for themes

 > ghostty theme

to quote yourself

> Some words and a code-block so u can copy-paste

(ghostty already gives code-block copy paste in all the overview sections its extremely excessive to do it for every single option)

searching "ghostty theme" does what you asked for perfectly as this page shows up where this header is right at the top
https://ghostty.org/docs/features/theme#listing-available-themes has a command to copy and paste to view all of the themes you have locally
right under that it has a link on how to view them online

 > For built-in themes, you can also view the theme list online at [iterm2-color-schemes](https://iterm2colorschemes.com/). This also includes a preview of each theme.

 > not as a first-time user. the docs should be understood by newbies

the config file is not necessarily meant for "newbies" in the future ghostty will have a complete GUI configuration option available to help less technical users. In the meantime we have a solid group of people dedicated to helping the newbies with questions like how to change the default shell

 > Also a basic config should be generated by default

ghostty already does this - it didn't in 1.0 and 1.1.0 if you already had a existing config after that feature was added it does not generate a config for you.

  > Oh yes, the best docs imo are:
    Vercel's many examples (that's how Next.js grew popular + Vercel's AI SDK)
    Tailwind CSS (see how it shows stuff visually)
    React.dev (recent site is gem)
    Vuejs.org (was a gem since v1 since Chinese devs don't have English as a first language so they make it very clear)

Ghostty does not need anywhere near the scale of documentation for its config as these listed above. They are all are so much more complex than simple key = value pairs that giving more in depth examples makes much more sense

> Lots of images will definitely take effort but will 100% take this project to the next level

most of the options you can configure / change have no clear visual indicator or a trivial to not need a image like changing your font / font-size 

 > What would be your suggestion then? A page detailing many common configuration options…?

This is a solid idea I suggest you made a new discussion otherwise it will get lost here since its not related to how to change the shell to fish

**Reply by deadcoder0904** [reply]
> https://ghostty.org/docs/features/theme#listing-available-themes has a command to copy and paste to view all of the themes you have locally

yeah, i did see that comment but i tried it but selecting it did nothing. it gave a select/option menu with no select/option so when i selected, it didn't work. i had to change the config file. seemed uninuititive since users are expected select/option to work like select/option. so it was defo confusing. 

not suggesting to do it for every single option but there's extreme amiguity. for example, take a look at FILE_LOCATION - https://ghostty.org/docs/features/theme#file-location (it uses lots of unknown variables with $ ... it should use simple Mac OS, Linux, Windows locations)

> the config file is not necessarily meant for "newbies" in the future ghostty will have a complete GUI configuration option available to help less technical users. In the meantime we have a solid group of people dedicated to helping the newbies with questions like how to change the default shell

well, if the docs answer it once, it'll save the most time for maintainers too. but yeah gui config seems exciting.

> Ghostty does not need anywhere near the scale of documentation for its config as these listed above. They are all are so much more complex than simple key = value pairs that giving more in depth examples makes much more sense

no, i'm talking about language, not examples. now i understand that all have `key = value` pairs for theme but i didn't understand where to paste `shell-integration = fish` since there is no filename. i think that would make it much easier to grasp but i haven't installed zsh in over a decade now but i really struggled with basic stuff with ghostty.... but i also didnt read as many words as i looked at code blocks, didnt understand what to put where, and then googled to find solutions.

> most of the options you can configure / change have no clear visual indicator or a trivial to not need a image like changing your font / font-size

that's true but some images are better than no images imo. even on homepage, a beautiful looking terminal would 100% be more effective. this is like a killer replacement for every terminal out there imo due to looks & speed but it doesn't show. speed can only be seen after using it but looks should be seen by everyone. altho yeah the ghostty animation looks amazing but i'd rather see a beautiful terminal with a simple config besides it. even the docs don't show any images. see how warp.dev does it even tho i switched from warp but their homepage shows rather than says. im only talking about images bcz even warp can be improved but ghostty is better. the only edge warp has over ghostty is ai features.

> This is a solid idea I suggest you made a new discussion otherwise it will get lost here since its not related to how to change the shell to fish

oh cool, the other maintainer asked so i mentioned it here. it was anyways just an idea to get more users/terminals because ghostty is ready for prime time for most normal people but does undersell itself through its homepage. im more talking from a marketing perspective, product can be improved a little bit mostly on onboarding other than that, its all snappy as hell. last time a terminal wowed me was a decade ago with zsh/ohmyzsh themes.

**Reply by pluiedev** [reply]
> it should use simple Mac OS, Linux, Windows locations

`XDG_CONFIG_HOME` is definitely a "simple" location to know - same as `/Library/Application Support` on macOS or `%AppData%` on Windows. While I wouldn't quite say we expect _everyone_ to know it, I would say there needs to be a prerequisite minimum level of technical understanding, especially if you're tinkering with a terminal.

Anyways, all of this is *extremely* off topic from the original question. If you wish to continue this thread about documentation further, please open a new discussion. Everyone who has previously participated here has gotten numerous emails/in-app notifications (assuming they haven't opted out of them) in a post we've already answered months ago, and we should stop disturbing them.
